src/protocols/secure_channel/CASESession.cpp

```diff
@@ -456,6 +442,22 @@ CHIP_ERROR CASESession::SendSigmaR2()
     uint8_t sr2k[kAEADKeySize];
     P256ECDSASignature tbsData2Signature;
 
+    uint32_t opcredsLen = 0;
+    uint8_t opcreds[kMaxCHIPCertLength];
```

Comment:
I am concerned of large stack usage on embedded devices (these will have 1K stacks or smaller). Could we scope-alloc these instead?

How much stack does this method use?

Reply:
I'll change it to alloc, instead of stack.
kMaxCHIPCertLength is 400 bytes. So it can have impact on stack usage for sure.